crates/ethereum-rpc/src/lib.rs

```diff
@@ -500,6 +514,91 @@ where
             })
     }
 
+    /// eth_sendRawTransactionSync RPC call implementation (EIP-7966)
+    /// - Send raw transaction to sequencer
+    /// - Subscribes to new block subscription
+    /// - Waits for transaction to be included in a block
+    /// - Returns the transaction receipt
+    /// - Timeout in milliseconds (default: 2_000ms, max: 60_000ms)
+    ///
+    /// Error codes:
+    /// - Code 4: Transaction not included within timeout period
+    /// - Code 32603: Node is running without subscriptions enabled or block channel closed while waiting on transaction inclusion
+    async fn eth_send_raw_transaction_sync(
+        &self,
+        data: Bytes,
+        timeout_ms: Option<u64>,
+    ) -> RpcResult<AnyTransactionReceipt> {
+        // eth_sendRawTransactionSync is not available if fullnode is running without `enable_subscriptions`
+        let mut block_rx = self
+            .l2_block_rx
+            .as_ref()
+            .ok_or_else(|| {
+                EthApiError::Unsupported(
+                    "Block subscriptions must be enabled for eth_sendRawTransactionSync",
+                )
+            })?
+            .resubscribe();
+
+        let hash: B256 = self
+            .ethereum
+            .sequencer_client
+            .as_ref()
+            .unwrap()
+            .eth_send_raw_transaction(data)
```

Comment:
I would say this should be on the sequencer level with a proper `eth_send_raw_transaction_sync`, so we don't spam sequencer every `eip_7966::calculate_timeout_ms(timeout_ms)` about the tx, and on the full-node side it's gonna be a direct:

```
    async fn eth_send_raw_transaction_sync(&self, data: Bytes) -> RpcResult<Option<Transaction>> {
        self.ethereum
            .sequencer_client
            .as_ref()
            .unwrap()
            .eth_send_raw_transaction_sync(data)
            .await
            .map_err(|e| match e {
                jsonrpsee::core::client::Error::Call(e_owned) => e_owned,
                _ => to_jsonrpsee_error_object("SEQUENCER_CLIENT_ERROR", e),
            })
    }
```